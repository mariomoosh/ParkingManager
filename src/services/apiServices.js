const SendRequest = async (url,method,postData = null,headers = null)=>{
    try{
        const requestOptions = {
            method,
            headers: {"Content-Type":"application/json"},
            body: postData !== null ? JSON.stringify(postData) : null
        }

        const response = await fetch(url,requestOptions);

        return await response.json();

    }catch(error){
        console.log(error);
        return null;
    };
}

const HTTPVERBS = {
    Get:"get",
    Post:"post",
    Put:"put",
    Delete:"delete"
}

export { SendRequest , HTTPVERBS };