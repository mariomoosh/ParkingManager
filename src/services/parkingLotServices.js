import appConfig from "../config/appConfig";
import * as apiServices from "./apiServices";

const getParkingLots = async ()=>{    
    const response = await apiServices.SendRequest(`${appConfig.ApiUrl}/ParkingLot`,apiServices.HTTPVERBS.Get,null);
    return response;
}

export {getParkingLots}