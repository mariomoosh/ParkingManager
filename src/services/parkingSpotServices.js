import appConfig from "../config/appConfig";
import * as apiServices from "./apiServices";

const getParkingSpotsByLotId = async (lotId)=>{    
    const response = await apiServices.SendRequest(`${appConfig.ApiUrl}/ParkingSpot?lotId=${lotId}`,apiServices.HTTPVERBS.Get,null);
    return response;
}

export {getParkingSpotsByLotId}