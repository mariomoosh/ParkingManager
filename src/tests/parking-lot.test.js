import * as lotServices from "../services/parkingLotServices";

jest.mock("../services/apiServices",()=>{
        return {
            SendRequest: jest.fn(()=>{return [{lotId:1,lotName:"dallas downtown"},{lotId:2,lotName:"dallas uptown"}]}),
            HTTPVERBS: jest.fn(()=>{return "Get"})
        }
    }
)

test("Should fetch parking lots",async ()=>{           
    const list  = await lotServices.getParkingLots();
    expect(list.length).toBe(2);
});
