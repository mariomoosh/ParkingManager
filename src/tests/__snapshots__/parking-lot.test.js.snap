// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Parking Spot doesn't have an image if empty 1`] = `
<div
  className="parkingSpot text-center spot-down spot-empty"
  onClick={[Function]}
>
  <label>
    1
  </label>
</div>
`;
